worldeditadditions/utils/parse/key_instance.lua

```diff
@@ -0,0 +1,69 @@
+--- A container for transmitting axis, sign pairs within parsing functions.
+-- @class
+local key_instance = {}
+key_instance.__index = key_instance
+key_instance.__name = "Key Instance"
+
+--- Creates a new Key Instance.
+-- @param: type: String: Key type (axis, direction, mirroring or error).
+-- @param: entry: String: The main content of the key.
+-- @param: sign: Int: The signed multiplier of the key (if any).
+-- @return: Key Instance: The new Key Instance.
+function key_instance.new(type,entry,sign)
+	if type == "error" then type = "err" end
+	local tbl = {type = type, entry = entry}
```

Comment:
Some additional type checking here wouldn't go amiss?

Reply:
If you mean `type()` checking then maybe. As far as `type` strings I left it open ended deliberately in case any other parsers want to use it and have other types to pass using key instances.

Reply:
I would suggest that it should be default to being strict. if we have another use case that uses different data types, we can revise later. I'd rather be too strict with types here than not strict enough.

Reply:
I really don't want to make another checking table... but I guess I could.

Reply:
Well that took a few more lines of code than I though but it's done.